### PR TITLE
feat(curriculum): update euler128 to more flexible problem, add solution

### PR DESCRIPTION
**Pull request by jeremylt**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes N/A

I updated Euler128 to test for two solutions and added a solution to the challenge.

### CODE REVIEW

**Review by Sboonny — APPROVED** [review]
Again, although closes nothing, contributing to #40896 

Thank you for this hard work LGTM 🎉.

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-128-hexagonal-tile-differences.md [hunk 1]
```diff
@@ -28,27 +28,79 @@ Find the 2000th tile in this sequence.
...
+      }
+    }
+  }
+})(NUM_PRIMES);
+
+function isPrime(num) {
+  if (num === 2) return true;
+  else if (num % 2 === 0) return false
+  else return PRIME_SEIVE[(num - 3) / 2];
+}
```
Comment (jeremylt) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-128-hexagonal-tile-differences.md:81]: For the ones that actually need a prime seive and are a bit slow, we can swap in this prime sieve that uses skips the multiples of two

**Review by moT01 — APPROVED** [review]
LGTM 🎉 Are you ready to go with this @jeremylt? No rush if you want to further improve it.

**Comment by jeremylt** [comment]
Good to go

**Pull request merged**